### ISSUE TITLE
Wait for multiple services

### ISSUE THREAD

**Issue by andrewshawcare** [vishnubob/wait-for-it#2]
It would be ideal if multiple services (i.e. multiple host:port combinations) could be declared such that the command is executed only after all dependent services are available.

**Comment by vishnubob** [comment]
I feel like this is better solved by either a top level script that orchestrates the logic of such a situation or chaining multiple `wait-for-it` commands together.

**Comment by andrewshawcare** [comment]
Yeah, I did that but that approach presents incorrect information.

For example:

```
./wait-for-it.sh service-a:1234 -- ./wait-for-it.sh service-b:1234 -- docker-entrypoint.sh
```

Logically, is `service-b` dependent on `service-a`? In my case it isn't (so they shouldn't be chained).

Structurally, there should be two options:

```
container -> {service-a, service-b} // set denotes mutually-exclusive dependencies
```

```
container -> [service-a, service-b] // list denotes dependency order
```

A chain provides a list mechanism and multiple `host:port` arguments provides a set mechanism.

**Comment by vishnubob** [comment]
A high-level script is your best option to solving this particular use case, as it allows you the flexibility you need in testing and interpreting the various states that arise in your specific scenario.  `wait-for-it` is intentionally designed to keep-it-simple, leaving the use-case you describe as an exercise for the reader.  Feel free to fork and tailor to your needs.

**Comment by jlordiales** [comment]
@vishnubob did you get to implement your use case? I was looking to do exactly the same

**Comment by vishnubob** [comment]
@jlordiales: i think you meant to ask @andrewshawcare, since he originally posted the issue.

**Comment by jlordiales** [comment]
Yes @vishnubob, sorry it was late :)

**Comment by jlordiales** [comment]
@andrewshawcare FWIW I implemented this on https://github.com/jlordiales/wait-for-it/blob/master/wait-for-it.sh (for my use case at least). You can specify multiple host:port pairs and the whole thing will fail as soon as the first host times out

**Comment by Forever-Young** [comment]
Didn't see this issue, also implemented this: https://github.com/Forever-Young/wait-for-it/commit/f9aa0c10c9e82e9ef3c2b744c7fa5f80aac7901a

**Comment by frastel** [comment]
Could this issue be closed as "won't fix" so it is clear that this high-level functionality shouldn't be integrated in this project?
As vishnubob already mentioned it makes totally sense to keep this script as simple as possible and to not patch this script for this reason.

**Comment by Forever-Young** [comment]
I guess it's enough to have some receipts just in the issue comments.

**Comment by clarketm** [comment]
There is a *python* version of [`wait-for-it`](https://github.com/clarketm/wait-for-it) that supports *waiting for multiple services*.

**e.g.**

```bash
$ wait-for-it \
--service www.google.com:80 \
--service www.bing.com:80 \
--service www.duckduckgo.com:80 \
-- echo "google, bing, and duckduckgo are up"
```

Here are the [docs](https://wait-for-it.readthedocs.io/en/latest/)!

**Comment by Feche-314** [comment]
There is also a shell script called [`docker-compose-wait`](https://github.com/ufoscout/docker-compose-wait/) which allows this required behaviour:

```
version: "3"

services:
  mongo:
    image: mongo:3.4
    hostname: mongo
    ports:
      - "27017:27017"

  postgres:
    image: "postgres:9.4"
    hostname: postgres
    ports:
      - "5432:5432"

  mysql:
    image: "mysql:5.7"
    hostname: mysql
    ports:
      - "3306:3306"

  mySuperApp:
    image: "mySuperApp:latest"
    hostname: mySuperApp
    environment:
      WAIT_HOSTS: postgres:5432, mysql:3306, mongo:27017

```

**Comment by codan84** [comment]
> There is also a shell script called [`docker-compose-wait`](https://github.com/ufoscout/docker-compose-wait/) which allows this required behaviour:
> 
> ```
> version: "3"
> 
> services:
>   mongo:
>     image: mongo:3.4
>     hostname: mongo
>     ports:
>       - "27017:27017"
> 
>   postgres:
>     image: "postgres:9.4"
>     hostname: postgres
>     ports:
>       - "5432:5432"
> 
>   mysql:
>     image: "mysql:5.7"
>     hostname: mysql
>     ports:
>       - "3306:3306"
> 
>   mySuperApp:
>     image: "mySuperApp:latest"
>     hostname: mySuperApp
>     environment:
>       WAIT_HOSTS: postgres:5432, mysql:3306, mongo:27017
> ```

Thanks for this tip, much better imo and not dependant on `bash` present in the image
